### ISSUE TITLE
the spacing does not match between the images?

### ISSUE THREAD

**Issue by yliliv** [MIC-DKFZ/nnUNet#622]
Hello!
I got the following error during preprocessing. What caused it? I look forward to your help.

![QQ截图20210503001948](https://user-images.githubusercontent.com/74352048/116820261-9ead2400-aba6-11eb-99b1-917d48c33224.png)

**Comment by FabianIsensee** [comment]
This is a warning, not an error.
Your segmentation file has a different geometry than the image it belongs to. It is your responsibility to check whether that is a problem or not (it can be, but does not have to be). 
You can try `nnUNet_plot_task_pngs [...] --use_raw` and inspect visually

**Comment by yliliv** [comment]
> This is a warning, not an error.
> Your segmentation file has a different geometry than the image it belongs to. It is your responsibility to check whether that is a problem or not (it can be, but does not have to be).
> You can try `nnUNet_plot_task_pngs [...] --use_raw` and inspect visually

Thank you for your answer, the above reason is that there is an incorrect data

**Comment by luoshiyong** [comment]
I get the same error! but my data is download from lits dataset.how can i solve this problem?

**Comment by sunshinelike** [comment]
Dear Sir,

 Can you give an example of use `nnUNet_plot_task_pngs [...] --use_raw' to solve the above referred problem.
Thank you very much!
Best regards.

**Comment by ykirchhoff** [comment]
Hi @luoshiyong and @sunshinelike,

if you still experience issues/have questions please open new issues and tag me there.

Best,
Yannick

**Comment by Vishu9519** [comment]
Hi, 
I'm getting a similar kind of warning, but I'm getting this after I've completed the training of a 2D model.

![image](https://github.com/MIC-DKFZ/nnUNet/assets/61423365/62850ff5-cfae-4ef8-bb35-d31136880201)

It trained the model for 1000 epochs, then predicted on the validation set, but I'm not getting what it is doing after that. It is trying to load files from the **gt_segmentations** folder, which was auto-created during the pre-processing step. 

Also, currently, it is training for 1000 epochs; is there a way to set custom epoch numbers? 
I'm trying to dive deep into the codes to understand the training, testing, and evaluation processes.

**Comment by FabianIsensee** [comment]
Take a look here: https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/dataset_format.md#supported-file-formats
nnU-Net expects to have a file for the spacing in case of tif images. If you don't have a specific voxel spacing it is save to ignore that

**Comment by Vishu9519** [comment]
Hi,
Thanks for the quick response. 

I've gone through the mentioned documentation file, and I understand that .tiff files need a spacing file in .json format.

I have the spacing files for the training images (in the nnUNet_raw/DatasetXXX../imagesTr/ )and labels (in the nnUNet_raw/DatasetXXX../lablesTr/). **This is not causing the issue.** 

Here is what is happening:

1. When I execute the **nnUNetv2_plan_and_preprocess -d ID --verify
_dataset_integrity** code, it generates and stores some files in the **"nnUNet_preprocessed/DatasetXXX"** folder, such as **gt_segmentations, nnUNetPlans_2d..** etc.
2. _**Inside**_ the auto-generated **gt_segmentations** folder, there is only .tiff files (same which are inside lablestr folder), and no corresponding .json file specifying the spacing parameter. This is producing the above-mentioned WARNING. 

Should I just **copy the .json files** from "nnUNet_raw/DatasetXXX/labelsTr/" to "nnUNet_Preprocessed/DatasetXXX/gt_segmentations"?

**Comment by FabianIsensee** [comment]
ah so this is then my bad, the json files need to be copied as well, thanks for the insight. I will fix this in the coming days
